package-build.el

```diff
@@ -1287,11 +1287,13 @@ ARCHIVE-ENTRY is destructively modified."
             (when commit
               (package-build-add-to-archive archive-entry :commit commit))
             (package-build--dump archive-entry
-                                 (package-build--entry-file-name archive-entry)))
-          (when package-build-write-melpa-badge-images
-            (package-build--write-melpa-badge-image
-             file-name
-             version package-build-archive-dir))
+                                 (package-build--entry-file-name archive-entry))
+            (when package-build-write-melpa-badge-images
```

Comment:
This `when` form needed to move inside the `let` so we'd have access to `archive-entry`.